lang/src/org/partiql/lang/planner/transforms/AstToLogicalVisitorTransform.kt

```diff
@@ -54,51 +54,46 @@ internal class AstToLogicalVisitorTransform(
             PartiqlLogical.build { limit(transformExpr(it), algebra, node.limit.metas) }
         } ?: algebra
 
-        return convertProjectionToBindingsToValues(node, algebra)
+        var expr = transformProjection(node, algebra)
+
+        // SELECT DISTINCT ...
+        expr = if (node.setq != null && node.setq is PartiqlAst.SetQuantifier.Distinct) {
+            return PartiqlLogical.build { call("filter_distinct", expr) }
+        } else expr
+
+        return expr
     }
 
-    private fun convertProjectionToBindingsToValues(node: PartiqlAst.Expr.Select, algebra: PartiqlLogical.Bexpr) =
+    private fun transformProjection(node: PartiqlAst.Expr.Select, algebra: PartiqlLogical.Bexpr) =
         PartiqlLogical.build {
-            bindingsToValues(
```

Comment:
Ty very much for cleaning this function up.

Reply:
https://i.imgur.com/tAgcJBD.jpg